corehq/apps/hqcase/tests/test_serialization.py

```diff
@@ -0,0 +1,92 @@
+import uuid
+from datetime import datetime
+
+from django.test import TestCase
+
+from casexml.apps.case.mock import CaseBlock, IndexAttrs
+
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.test_utils import create_and_save_a_case
+
+from ..api.core import serialize_case
+from ..utils import submit_case_blocks
+
+
+class TestAPISerialization(TestCase):
+    domain = 'test-update-cases'
+    maxDiff = None
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(cls.domain)
+        cls.case_accessor = CaseAccessors(cls.domain)
+
+        cls.parent_case_id = str(uuid.uuid4())
+        case_id = str(uuid.uuid4())
+        xform, cases = submit_case_blocks([
+            CaseBlock(
+                case_id=cls.parent_case_id,
+                case_type='player',
+                case_name='Elizabeth Harmon',
+                external_id='1',
+                owner_id='methuen_home',
+                create=True,
+                update={
+                    'sport': 'chess',
+                    'rank': '1600',
+                    'dob': '1948-11-02',
```

Comment:
I'm curious to see how decimals are serialized. A case block value of `'5.0'` will result in a case property value of `Decimal('5.0')`. What will the Elasticsearch data type be? What will the API return?

Reply:
It turns out, that "Decimal" thing is more complicated. I posted on Slack:

> When testing data types in payloads for data forwarding, I discovered that case properties that look like decimals are cast as `Decimal` in the Couch backend, but not the SQL backend. (All other case properties remain `str`.) I didn't expect that. ([The test is here](https://github.com/dimagi/commcare-hq/blob/af1e14c03c0476ed4385e53193220d5efff8305e/corehq/motech/repeaters/tests/test_payload_generators.py#L150-L166).)

Reply:
That's surprising, though perhaps more surprising is that this hasn't popped up loudly enough to have been addressed already.  I'll add a test to see.